pkg/network/config/sysctl/sysctl.go

```diff
@@ -0,0 +1,91 @@
+// +build linux
+
+package sysctl
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// String represents a string sysctl
+type String struct {
+	*sctl
+	v string
+}
+
+// NewString creates a new sysctl.String
+//
+// `procRoot` points to the procfs root, e.g. /proc
+// `sysctl` is the path for the sysctl, e.g. /proc/sys/<sysctl>
+// `cacheFor` caches the sysctl's value for the given time duration;
+// `0` disables caching
+func NewString(procRoot, sysctl string, cacheFor time.Duration) *String {
+	return &String{sctl: newSCtl(procRoot, sysctl, cacheFor)}
+}
+
+// Get gets the current value of the sysctl
+func (s *String) Get() (string, error) {
+	v, updated, err := s.sctl.Get()
+	if err == nil && updated {
+		s.v = v
+	}
+
+	return s.v, err
+}
+
+// Int represents an int sysctl
+type Int struct {
+	*sctl
+	v int
+}
+
+// NewInt creates a new sysctl.Int
+//
+// `procRoot` points to the procfs root, e.g. /proc
+// `sysctl` is the path for the sysctl, e.g. /proc/sys/<sysctl>
+// `cacheFor` caches the sysctl's value for the given time duration;
+// `0` disables caching
+func NewInt(procRoot, sysctl string, cacheFor time.Duration) *Int {
+	return &Int{sctl: newSCtl(procRoot, sysctl, cacheFor)}
+}
+
+// Get gets the current value of the sysctl
+func (i *Int) Get() (int, error) {
+	v, updated, err := i.sctl.Get()
+	if err == nil && updated {
+		i.v, err = strconv.Atoi(v)
+	}
+
+	return i.v, err
+}
+
+type sctl struct {
+	ttl      int64
```

Comment:
Any reason not to leave this a `time.Duration`? I think using the real type is much clearer.